### DISCUSSION TITLE
How to add "recipients" for DIDComm JWE

### DISCUSSION THREAD

**Discussion by amlord** [panva/jose#311] [Q&A] [answered]
Hello & thanks for the great package!

I've been looking at adding DIDComm messaging to our project, and have been using the test vectors in the spec to verify [signed messages](https://identity.foundation/didcomm-messaging/spec/#c2-didcomm-signed-messages), and [decrypt encrypted messages](https://identity.foundation/didcomm-messaging/spec/#c3-didcomm-encrypted-messages).

I've also managed to encrypt a message fine, but am struggling to work out how the `recipients` property can be added. I've looked at the docs and searched the repo, and came across [this discussion](https://github.com/panva/jose/discussions/21#discussioncomment-210418) from an older version of the package. There doesn't seem to be an option to add recipients in the latest version though, so would someone be able to point me in the right direction on how this should be achieved?

**Comment by panva** [comment, marked as answer]
General JWE Encryption for multiple recipients is not supported in v3.x/v4.x

![image](https://user-images.githubusercontent.com/241506/140740834-0a9fe932-cc24-4847-81fd-dfb97ec348f2.png)

**Reply by amlord** [reply]
oh, ok thanks - so should I use an older version, or maybe there's a really good reason for this?

**Reply by panva** [reply]
You can use v2.x for this niche use case if your target JavaScript runtime is Node.js.

**Reply by amlord** [reply]
that's great, thanks again! 👍🏻

**Reply by panva** [reply]
I'll be adding general encrypt to v4.x later this week.

**Reply by panva** [reply]
https://github.com/panva/jose/releases/tag/v4.2.0

**Reply by amlord** [reply]
oh, wow - that's great, thanks; I've spent most of this week reading up & wraaping DID resolution of public keys for signing & verification (which I'm just about there with), so this comes at a really good time as I'll be getting onto the encrypt / decrypt part next week...

you're a  :star: !
